src/irmin-pack/ext.ml

```diff
@@ -17,29 +17,24 @@
 open! Import
 module IO = IO.Unix
 
-module Maker
-    (V : Version.S)
-    (Config : Conf.S)
-    (Node : Irmin.Node.Maker)
-    (Commit : Irmin.Commit.Maker) =
-struct
+module Maker (V : Version.S) (Config : Conf.S) = struct
```

Comment:
I was hoping that the `Maker` level would be collapsed by moving `V` and `Config` to `Schema`.

Is this doable?

Reply:
I had a go at this. I'm unsure as to whether it works out to be better overall -- neither the config nor the version impacts the API of the store, so there are a few call-sites in the code where we want to be generic over one or the other. Regardless, the diff is quite separate from this one so I'll package it in a second PR and we can discuss from there.